src/useImageDimensions.ts

```diff
@@ -5,51 +5,55 @@ export interface URISource {
   uri: string
 }
 
-class Dimensions {
-  width: number
-  height: number
-  constructor(width: number, height: number) {
-    this.width = width
-    this.height = height
-  }
-  /**
-   * width to height ratio
-   */
-  get aspectRatio() {
-    return this.width / this.height
-  }
+export type Source = ImageRequireSource | URISource
```

Comment:
Why do you make one more type and export it?

Reply:
If someone wants to build a hook that extends on this hook, it's nice to be able to have one type that represents any input. That way it can easily be used as the input value to a function and just passed on, e.g:

```typescript
function useHeaderHeight (logo: Source, background: Source): number {
  // ...
}
```